server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -236,7 +237,23 @@ private synchronized void updateSkipUnavailable(String clusterAlias, Boolean ski
 
     @Override
     protected void updateRemoteCluster(String clusterAlias, Settings settings) {
-        updateRemoteCluster(clusterAlias, settings, noopListener);
+        if (remoteClusters.containsKey(clusterAlias) == false) {
+            CountDownLatch latch = new CountDownLatch(1);
+            updateRemoteCluster(clusterAlias, settings, ActionListener.wrap(latch::countDown));
+
+            try {
+                // Wait 10 seconds for a new connection. We must use a latch instead of a future because we
+                // are on the cluster state thread and our custom future implementation will throw an
+                // assertion.
+                if (latch.await(10, TimeUnit.SECONDS) == false) {
+                    logger.warn("failed to connect to updated remote cluster {} within {}", clusterAlias, TimeValue.timeValueSeconds(3));
```

Comment:
failed to connect to *new* remote cluster

also note that you're using 3 seconds in the message here 🗡